### PR TITLE
fix: redis diagnostics panic

### PR DESCRIPTION
**Pull request by decibelcooper**
# Context
I discovered a panic that occurs when a redis-backend workflow instance expires followed by fetching workflow instances through the `diag.Backend` interface.
```
panic: interface conversion: interface {} is nil, not string
```

# Changes
* check for validity of string assertion, and skip if not a string
* piggy-back on existing expire test to prevent regression and ensure that the diag result is sane

### CODE REVIEW

**Comment by decibelcooper** [comment]
FYI in with regards to https://github.com/cschleiden/go-workflows/issues/280, this is not an error scenario by my judgement.  Instead, this is a benign case where we GET something that has expired but hasn't yet been removed from the index.

**Comment by cschleiden** [comment]
Oh I see, so the expired instance is still in the `instances-by-creation` set because no other workflow instance has ended so `expireCmd` hasn't been called again, **but** the expired instance's data keys have already `EXPIRE`d.. 🤔

File: backend/test/backend.go [hunk 1]
```diff
@@ -5,10 +5,12 @@ import (
 
 	"github.com/cschleiden/go-workflows/backend"
 	"github.com/cschleiden/go-workflows/backend/history"
+	"github.com/cschleiden/go-workflows/diag"
 )
 
 type TestBackend interface {
 	backend.Backend
+	diag.Backend
```
Comment (cschleiden) [inline backend/test/backend.go:13]: I'd prefer if we kept this separate and added a new set of tests cases for diag backends instead

Comment (decibelcooper) [reply backend/test/backend.go:13]: sure I can make a change for that.

Comment (decibelcooper) [reply backend/test/backend.go]: removed in favor of simply asserting the implementation type in the test

**Review by cschleiden — APPROVED** [review]
LGTM! :shipit:

**Pull request merged**